source/volume_of_fluid/reconstruct.cc

```diff
@@ -0,0 +1,488 @@
+      }
+
+    initial_solution.compress(VectorOperation::insert);
+
+    sim.compute_current_constraints();
+    sim.current_constraints.distribute(initial_solution);
+
+    // solution.block(volume_of_fluid_blockidx) = initial_solution.block(volume_of_fluid_blockidx);
+    solution.block(volume_of_fluidN_blockidx) = initial_solution.block(volume_of_fluidN_blockidx);
+    solution.block(volume_of_fluidLS_blockidx) = initial_solution.block(volume_of_fluidLS_blockidx);
+
+    sim.computing_timer.exit_section();
+  }
+
+
+  template <>
+  void VolumeOfFluidHandler<3>::update_volume_of_fluid_normals (const VolumeOfFluidField<3> /*field*/,
+                                                                LinearAlgebra::BlockVector &/*solution*/)
+  {
+    Assert(false, ExcNotImplemented());
+  }
+
+  template <>
+  void VolumeOfFluidHandler<2>::update_volume_of_fluid_composition (const typename Simulator<2>::AdvectionField composition_field,
+                                                                    const VolumeOfFluidField<2> volume_of_fluid_field,
+                                                                    LinearAlgebra::BlockVector &solution)
+  {
+    const int dim = 2;
+
+    LinearAlgebra::BlockVector initial_solution;
+
+    sim.computing_timer.enter_section("  Compute VolumeOfFluid compositions");
+
+    initial_solution.reinit(sim.system_rhs, false);
+
+    // Normal holding vars
+    Point<dim> uReCen;
+
+    for (unsigned int i=0; i<dim; ++i)
+      uReCen[i] = 0.5;
+
+    const FiniteElement<dim> &system_fe = this->get_fe();
+
+    std::vector<types::global_dof_index> local_dof_indicies (system_fe.dofs_per_cell);
+
+    const FEVariable<dim> &volume_of_fluid_var = volume_of_fluid_field.volume_fraction;
+    const unsigned int volume_of_fluid_c_index = volume_of_fluid_var.first_component_index;
+    const unsigned int volume_of_fluid_ind
+      = system_fe.component_to_system_index(volume_of_fluid_c_index, 0);
+
+    const FEVariable<dim> &volume_of_fluidN_var = volume_of_fluid_field.reconstruction;
+    const unsigned int volume_of_fluidN_c_index = volume_of_fluidN_var.first_component_index;
+
+    const unsigned int base_element = composition_field.base_element(this->introspection());
+    const std::vector<Point<dim> > support_points = system_fe.base_element(base_element).get_unit_support_points();
+
+    for (auto cell : this->get_dof_handler().active_cell_iterators ())
+      {
+        if (!cell->is_locally_owned())
+          continue;
+
+        cell->get_dof_indices (local_dof_indicies);
+        const double cell_volume_of_fluid = solution(local_dof_indicies[volume_of_fluid_ind]);
+
+        Tensor<1, dim, double> normal;
+
+        for (unsigned int i=0; i<dim; ++i)
+          normal[i] = solution(local_dof_indicies[system_fe
+                                                  .component_to_system_index(volume_of_fluidN_c_index+i, 0)]);
+
+        Tensor<1, dim, double> nnormal;
+        double normall1n = 0.0;
```

Comment:
```suggestion
        double normal_l1_norm = 0.0;
```
This took me a minute to figure out, and everyone else who will read this code as well.